File: solutions/java/resistor-color-duo/2/src/main/java/ResistorColorDuo.java
import java.util.List;
import java.util.Arrays;

class ResistorColorDuo {
    private final List<String> COLORS_LIST = Arrays.asList("black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white");
    
    int value(String[] colors) {
        if(colors.length < 2) {
            throw new UnsupportedOperationException("At least two colors have to be given.");
        }
        int colorOneValue = COLORS_LIST.indexOf(colors[0]);
        int colorTwoValue = COLORS_LIST.indexOf(colors[1]);
        
        return colorOneValue * 10 + colorTwoValue;
    }
}

File: solutions/python/resistor-color-duo/2/resistor_color_duo.py
"""
This module provides functionality to calculate resistor values based on color codes.
"""

RESISTOR_COLORS = (
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
)
"""
Tuple of color codes used for calculating resistor values.
Each color corresponds to a number from 0 to 9.
"""

color_code = RESISTOR_COLORS.index
"""
Function to get the index of a given color in RESISTOR_COLORS tuple.
"""


def value(colors):
    """
    Calculate the numerical value of a resistor given its color bands.

    :param colors: A list of the first two color bands of the resistor.
    :return: The numerical value as an integer.
    """
    first_color, second_color = colors[:2]
    return color_code(first_color) * 10 + color_code(second_color)
